include/envoy/server/transport_socket_config.h

```diff
@@ -63,6 +68,17 @@ class TransportSocketFactoryContext {
    * @return the server-wide stats store.
    */
   virtual Stats::Store& stats() PURE;
+
+  /**
+   * Pass an init manager to register dynamic secret provider.
+   * @param init_manager instance of init manager.
+   */
+  virtual void setInitManager(Init::Manager& init_manager) PURE;
+
+  /**
+   * @return the instance of init manager.
+   */
+  virtual Init::Manager& initManager() PURE;
```

Comment:
since initManager is assigned not in the constructor, we need to return raw pointer so it can return nullptr to indicate if it is assigned or not.